pkg/grpc/interceptor/auth/client.go

```diff
@@ -1,4 +1,4 @@
-package apikey_interceptor
+package auth_interceptor
```

Comment:
Can we split this in 2 implementations?

Making an interface like this:
```
type ClientInterceptor interface {
  GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error)
  RequireTransportSecurity() bool
}
```
And implementing the `PSKInterceptor` and `APIKeyInterceptor` as separate implementations